Make parameterized ArgumentSource annotations repeatable (#3787)
junit-team/junit5: documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

(like built-in providers such as `{ValueArgumentsProvider}` or `{CsvArgumentsProvider}`),
you have the possibility to extend the `{AnnotationBasedArgumentsProvider}` class.

[[writing-tests-parameterized-repeatable-sources]]
===== Multiple sources using repeatable annotations

> Repeatable annotations allows you a convenient way to provide multiple sources from
> different providers whenever it fits.

Reviewer: ```suggestion
Repeatable annotations provide a convenient way to specify multiple sources from
different providers.
```

Author: Rephrased accordingly in commit f8d041d

[source,java,indent=0]
----
include::{testDir}/example/ParameterizedTestDemo.java[tags=repeatable_annotations]
----

Following the above parameterized test, a test case will run for each argument:

----
[1] foo
[2] bar
----

> Annotations that support repeatable capability:

Reviewer: ```suggestion
The following annotations are repeatable:
```

Author: Rephrased accordingly in commit f8d041d
